cmd/skopeo/unshare_linux.go

```diff
@@ -32,3 +34,13 @@ func maybeReexec() error {
 	}
 	return nil
 }
+
+func needReexec(imageNames []string) error {
```

Comment:
(Now that this is being revisited: `needReexec` sounds as if it is returning a boolean.  I can’t think of a _good_ replacement but at least a cumbersome `reexecIfNecessaryForImages` would 1) clearly indicate an action, and 2) explain the obscure thing we are dealing with, for future readers who will have no idea that this is necessary and why. Or maybe some simpler name, with a doc string that clearly describes what is going on.)

Reply:
Next commit uses the cumbersome name so that it embed the action purpose for the caller.